### PR TITLE
Virtual Mouse Input

### PR DESCRIPTION
**Pull request by mcpalmer1980**
For https://github.com/pygame/pygame/issues/1210

It seems that virtual mouse input at least partially works on OSX, but on my Linux Mint 18.3 it definitely doesn't. These changes do fix it.

I'm not sure how to write an automated test for input code, but I can upload an example that works with this branch but not in any of the pygame 2.0 dev builds. It creates a huge surface showcasing all your fonts and lets you scroll vertically through them, entering and exiting virtual mode with a left click.

### CODE REVIEW

**Comment by mcpalmer1980** [comment]
Tried a script that uses virtual input mode for dev4 on Windows and I got no input beyond window edges. I wanted to try my fork but instructions for building on Windows asks for msvc, which asked for 7gb install. Don't have that much space available for Windows right now.
I'm pretty sure my changes would fix the problem.

I think I'll upload the example I used, which is a bit messy, and a bare-bones one. If anyone thinks one is useful and worthy of inclusion, I'll clean it up. 

In the mean time, I'm open for any suggestions regarding a test for mouse.c and/or my changes. Other than manually setting position and then getting it, which isn't guaranteed to simulate real world mouse movement, I'm at a loss.

**Comment by illume** [comment]
I'm seeing this error on Mac clang:
```
src_c/event.c:140:19: error: unused variable 'inputEvent' [-Werror,-Wunused-variable]
        SDL_Event inputEvent[2];
```

**Comment by illume** [comment]
I like the examples :)
We don't have a font viewer type example already. I wonder if we can add the relative mouse stuff to the eventlist.py status display.

In the chat it sounded like you have already tested quite a lot of things. Let me know if there's anything in specific that you'd like me to test? I have mac/windows/linux machines set up.

For automated testing, just calling the code provides good value. It shows that the code paths do not segfault on 200+ different OS/python combinations. It also shows that the argument processing works. With runtime instrumentation in things like Valgrind, and ASan memory testers it allows those tools to test the code paths... and that the SDL code itself doesn't crash on different platforms. So just calling `pygame.mouse.set_visible()` in a test would be good. Then testing if you set it to visible, if the get_visible function returns that it's visible [and the opposite].

ps. We have 'interactive' tests. `python3 -m pygame.tests --interactive`, which allow you to ask the user a question. We don't have many of these, but it can be a good way to put into code things which are hard/impossible to test automatically.
- write the test a human should do as text. "Use examples.virtualmouse and see what the coordinates printed are XXX if you do ZZZ..."
- ask them a Y/N question if it worked.

**Comment by illume** [comment]
There already is a basic test for get_visible set_visible.

File: src_c/event.c [hunk 1]
```diff
@@ -137,10 +137,7 @@ pg_event_filter(void *_, SDL_Event *event)
 #pragma PG_WARN(Add event blocking here.)
 
     else if (type == SDL_KEYDOWN) {
-#ifdef WIN32
```
Comment (illume) [inline src_c/event.c]: here is the removed ifdef.

File: examples/font_viewer.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env python3
+
+import pygame as pg
+from pygame.constants import *
+
+class fontviewer:
```
Comment (illume) [inline examples/font_viewer.py]: `class fontviewer` should be `class FontViewer` to be pep8 friendly.

File: examples/font_viewer.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		self.clock = pg.time.Clock()
+		self.y_offset = 0
+		self.grabbed = False
+		self.Render('&N abcDEF789', size=48, color = (0,0,0), back_color = (255,255,255))
+		self.Display()
+		#viewer.Save()
+		#pg.mouse.set_visible(not self.grabbed)
+		#pg.event.set_grab(self.grabbed)
+
+	def Render(self, text = 'A display of font &N', **dparams):
```
Comment (illume) [inline examples/font_viewer.py]: methods should be lower case for pep8 friendliness

**Pull request closed without merging**